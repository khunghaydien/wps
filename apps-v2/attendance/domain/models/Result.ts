export type ResultSuccess<Value = void> = { result: true; value: Value };

export type ResultFailed<Reason> = {
  result: false;
  reason: Reason;
};

export type Result<Reason, Value = unknown> =
  | ResultSuccess<Value>
  | ResultFailed<Reason>;

export const REASON = {
  UNEXPECTED: 'unexpected',
  USER_INDUCED: 'userInduced',
  NO_RECORD: 'noRecord',
  EXISTED_INVALID_REQUEST: 'existedInvalidRequest',
  EXISTED_SUBMITTING_REQUEST: 'existedSubmittingRequest',
} as const;

export type ReasonMap = typeof REASON;

export type UnexpectedReason = ReasonMap['UNEXPECTED'];

export type UserInducedReason = ReasonMap['USER_INDUCED'];

export type NoRecord = ReasonMap['NO_RECORD'];

export type NotSubmittedReason =
  | ReasonMap['EXISTED_INVALID_REQUEST']
  | ReasonMap['EXISTED_SUBMITTING_REQUEST'];
