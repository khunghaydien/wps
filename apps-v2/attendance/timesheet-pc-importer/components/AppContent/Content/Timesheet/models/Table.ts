import * as ReactGrid from '@silevis/reactgrid';
import { Column } from '@silevis/reactgrid';

export type ErrorsCell = ReactGrid.Cell & {
  type: 'errors';
  errors?: string[];
};

export type AttTimeCell = ReactGrid.Cell & {
  type: 'attTime';
  text: string;
  value: number;
  placeholder?: string;
  renderer?: (text: string) => React.ReactNode;
};

export type Cell =
  | ReactGrid.CheckboxCell
  | ReactGrid.HeaderCell
  | ReactGrid.TextCell
  | AttTimeCell
  | ErrorsCell
  | ReactGrid.DropdownCell;

export type CellId = ReactGrid.Id;

export { type Column };

export type Row = ReactGrid.Row<Cell>;

export type CellChange = ReactGrid.CellChange<Cell>;
