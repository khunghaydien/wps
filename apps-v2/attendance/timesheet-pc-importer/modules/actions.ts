/**
 * 使用できる Module の Action です。
 *
 * Module が common や Local などの寄せ集めになっており、
 * 使用するべき Action を理解するのが困難になったため集約させました。
 */
export * as common from './common/actions';
export { actions as timesheet } from './timesheet';
export * as widgets from './widgets/actions';
