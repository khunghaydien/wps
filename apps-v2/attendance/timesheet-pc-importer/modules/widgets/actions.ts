export { actions as ProxyEmployeeSelectDialog } from '../../../../../widgets/dialogs/ProxyEmployeeSelectDialog/modules/ui';
