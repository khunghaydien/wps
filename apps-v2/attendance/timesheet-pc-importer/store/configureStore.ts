/* eslint-disable global-require, import/newline-after-import */
import { applyMiddleware, compose, createStore, Middleware } from 'redux';
import logger from 'redux-logger';
// 本来は使いたくないが
// Global Container の社員切替が使用しているので
// しかたなく導入している
import thunkMiddleware from 'redux-thunk';

import rootReducer, { State } from '../modules';

const PRODUCTION = process.env.NODE_ENV === 'production';
const TEST = process.env.TEST === 'test';

/* eslint-disable no-underscore-dangle */
const composeEnhancers =
  !PRODUCTION && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    : compose;

export default function configureStore(initialState?: State) {
  const middleWares: Middleware[] = [
    // 本来は使いたくないが
    // Global Container の社員切替が使用しているので
    // しかたなく導入している
    thunkMiddleware,
  ];
  if (!PRODUCTION && !TEST) {
    middleWares.push(logger);
  }
  const store = createStore(
    rootReducer,
    initialState, // Redux DevTools Extension
    composeEnhancers(applyMiddleware(...middleWares))
  );

  if (!PRODUCTION && module.hot) {
    module.hot.accept('../modules', () => {
      // eslint-disable-next-line @typescript-eslint/no-var-requires
      const newRootReducer = require('../modules').default;
      store.replaceReducer(newRootReducer);
    });
  }

  return store;
}
