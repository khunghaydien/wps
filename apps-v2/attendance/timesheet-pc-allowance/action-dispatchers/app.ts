import { getUserSetting } from '../../../commons/actions/userSetting';
import DateUtil from '../../../commons/utils/DateUtil';
import UrlUtil from '../../../commons/utils/UrlUtil';

import { actions as dailyallowanceActions } from '../modules/entities/dailyallowance';

const initialize = () => (dispatch) => {
  const params = UrlUtil.getUrlQuery();
  const startDate = params ? params.startDate : DateUtil.nowAsISO8601();
  const endDate = params ? params.endDate : DateUtil.nowAsISO8601();
  const targetEmployeeId = params ? params.targetEmployeeId : null;

  return Promise.all([
    dispatch(getUserSetting()),
    dispatch(dailyallowanceActions.fetch(startDate, endDate, targetEmployeeId)),
  ]);
};

const openPrintDialog = () => () => {
  window.print();
};

// eslint-disable-next-line import/prefer-default-export
export const actions = {
  initialize,
  openPrintDialog,
};
