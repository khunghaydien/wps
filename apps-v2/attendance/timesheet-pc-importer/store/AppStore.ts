import { Action, Store } from 'redux';
import { ThunkDispatch } from 'redux-thunk';

import { State } from '@attendance/timesheet-pc-importer/modules';

export type AppState = State;
export type AppStore = Store<State>;
export type AppDispatch = ThunkDispatch<unknown, void, Action<string>>;
