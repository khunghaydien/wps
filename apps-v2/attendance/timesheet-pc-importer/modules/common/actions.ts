import * as app from '@commons/actions/app';

export { app };
export { actions as proxyEmployeeInfo } from '@commons/modules/proxyEmployeeInfo';
export { showToast, actions as toast } from '@commons/modules/toast';
