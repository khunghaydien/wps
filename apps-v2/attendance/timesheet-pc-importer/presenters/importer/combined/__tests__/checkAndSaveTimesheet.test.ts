import configureMockStore from 'redux-mock-store';
import thunkMiddleware from 'redux-thunk';

import { confirm } from '@commons/actions/app';

import { REASON } from '@attendance/domain/models/Result';

import presenter from '../checkAndSaveTimesheet';
import { AppStore } from '@attendance/timesheet-pc-importer/store/AppStore';

jest.mock('@commons/actions/app', () => {
  const original = jest.requireActual('@commons/actions/app');
  return {
    ...original,
    __esModules: true,
    confirm: jest.fn(),
  };
});

jest.mock('@commons/modules/toast', () => ({
  __esModules: true,
  showToast: (...args) => ({
    type: 'TEST:SHOW_TOAST',
    payload: args,
  }),
}));

const mockStore = configureMockStore([thunkMiddleware]);

beforeEach(() => {
  jest.clearAllMocks();
});

describe('confirmSubmittingWithoutErrorRecords()', () => {
  it('should return true if errors are nothing.', () => {
    // Arrange
    const store = mockStore({
      timesheet: {
        records: new Map([['recordDate', {}]]),
      },
    });

    // Act
    presenter(store as AppStore)().confirmSubmittingWithoutErrorRecords(null);

    // Assert
    expect(store.getActions()).toMatchSnapshot();
  });
  it('should call confirm with errors.', async () => {
    // Arrange
    const store = mockStore({
      timesheet: {
        records: new Map([
          [
            'recordDate',
            {
              checked: true,
              errors: 'error',
            },
          ],
        ]),
      },
    });
    (confirm as jest.Mock).mockImplementation((...args) => (dispatch) => {
      dispatch({
        type: 'TEST:CONFIRM',
        payload: args,
      });
      return 'result';
    });

    // Act
    const result = await presenter(
      store as AppStore
    )().confirmSubmittingWithoutErrorRecords(null);

    // Assert
    expect(result).toEqual('result');
    expect(store.getActions()).toMatchSnapshot();
  });
});

describe('complete()', () => {
  it('should do if result is true.', () => {
    // Arrange
    const store = mockStore({});

    // Act
    presenter(store as AppStore)().complete({ result: true, value: undefined });

    // Assert
    expect(store.getActions()).toMatchSnapshot();
  });
  it('should do if result is false.', () => {
    // Arrange
    const store = mockStore({});

    // Act
    presenter(store as AppStore)().complete({
      result: false,
      reason: REASON.USER_INDUCED,
    });

    // Assert
    expect(store.getActions()).toMatchSnapshot();
  });
});

describe('error()', () => {
  it('should do.', () => {
    // Arrange
    const store = mockStore({});

    // Act
    presenter(store as AppStore)().error({
      message: 'Error Test',
    });

    // Assert
    expect(store.getActions()).toMatchSnapshot();
  });
});

describe('start()', () => {
  it('should do.', () => {
    // Arrange
    const store = mockStore({});

    // Act
    presenter(store as AppStore)().start();

    // Assert
    expect(store.getActions()).toMatchSnapshot();
  });
});

describe('finally()', () => {
  it('should do.', () => {
    // Arrange
    const store = mockStore({});

    // Act
    presenter(store as AppStore)().finally();

    // Assert
    expect(store.getActions()).toMatchSnapshot();
  });
});
