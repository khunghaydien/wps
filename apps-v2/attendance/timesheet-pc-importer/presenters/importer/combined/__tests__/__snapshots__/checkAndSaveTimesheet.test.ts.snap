// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`complete() should do if result is false. 1`] = `Array []`;

exports[`complete() should do if result is true. 1`] = `Array []`;

exports[`confirmSubmittingWithoutErrorRecords() should call confirm with errors. 1`] = `
Array [
  Object {
    "type": "LOADING_END",
  },
  Object {
    "payload": Array [
      "There are the following notes.Are you sure to apply to the timesheet?
・There is errors that you selected date.If you apply with this data, The data that has error will not applied.",
    ],
    "type": "TEST:CONFIRM",
  },
  Object {
    "type": "LOADING_START",
  },
]
`;

exports[`confirmSubmittingWithoutErrorRecords() should return true if errors are nothing. 1`] = `Array []`;

exports[`error() should do. 1`] = `
Array [
  Object {
    "payload": Object {
      "errorCode": undefined,
      "isContinuable": true,
      "isFunctionCantUseError": false,
      "message": "APIエラー - Error Test",
      "problem": "Error Test",
      "stackTrace": "",
      "type": "APIエラー",
    },
    "type": "CATCH_API_ERROR",
  },
]
`;

exports[`finally() should do. 1`] = `
Array [
  Object {
    "type": "LOADING_END",
  },
]
`;

exports[`start() should do. 1`] = `
Array [
  Object {
    "type": "LOADING_START",
  },
]
`;
